src/service-worker-utils.ts

```diff
@@ -5,7 +5,9 @@ import { uiLogger } from './lib/logger.js'
  */
 export async function registerServiceWorker (): Promise<ServiceWorkerRegistration> {
   const log = uiLogger.forComponent('register-service-worker')
-  const swRegistration = await navigator.serviceWorker.register(new URL(/* webpackChunkName: "sw" */'sw.js', import.meta.url))
+  const swRegistration = await navigator.serviceWorker.register(new URL('ipfs-sw-sw.js', import.meta.url), {
+    type: 'module'
```

Comment:
We might have trouble with Firefox.
https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker#browser_compatibility

Have you tried setting the esbuild bundle format to `iife`? https://esbuild.github.io/api/#format-iife

Reply:
Fixed in https://github.com/ipfs/service-worker-gateway/pull/392/commits/4225e61aa11daa4eaa7c65a433c014571901a686